File: Solutions/Python/reverse_words_in_a_string.py
"""
    [MEDIUM]
    151. Reverse Words in a String

    Concepts:
    - string

    Stats:
        Runtime | 0 ms      [Beats 100%]
        Memory  | 17.80 MB  [Beats 45.94%]
"""

class Solution:
    def reverseWords(self, s: str) -> str:
        return " ".join(s.split()[::-1])

File: Solutions/Rust/reverse_words_in_a_string.rs
/*
    [MEDIUM]
    151. Reverse Words in a String

    Concepts:
    - string

    Stats:
        Runtime | 0 ms      [Beats 100%]
        Memory  | 2.12 MB   [Beats 87.20%]
*/
impl Solution {
    pub fn reverse_words(s: String) -> String {
        return s
            .split_whitespace()
            .rev()
            .collect::<Vec<&str>>()
            .join(" ");
    }
}